datasetIngestor/getLocalFileList.go

```diff
@@ -261,22 +191,34 @@ Do you want to keep the link in dataset or skip it (D(efault)/k(eep)/s(kip) ?`,
 			return err
 		})
 
-		if e != nil {
-			log.Fatal("Fatal error:", e)
+		if err != nil {
+			return []Datafile{}, time.Time{}, time.Time{}, "", 0, 0, fmt.Errorf("file walk returned error: %v", err)
 		}
 	}
 	// spin.Stop()
 	return fullFileArray, startTime, endTime, owner, numFiles, totalSize, err
```

Comment:
I meant why not returning 
fullFileArray, startTime, endTime, owner, numFiles, totalSize, **skippedLinks, illegalFileNames**, err

Reply:
skippedLinks and illegalFilenames are both handled by the callback functions, so if counting is important, they can do it. That's how I written up this PR as well. 

Another minor reason is that having too many returns also makes the code less accessible. 

Although, while I feel like the way I set it up is pretty interesting, it's also not the easiest to understand for a beginner in Go. 

I'm against changing it, but if you're convinced, just let me know and I'll change it.

Reply:
the counting can still happen from within the callbacks as long as they return it. I am fine with your implementation, I tend to avoid changes "in place". Maybe a way in between is to pass a struct to GetFiles rather than a callback and this struct has an internal counter? But otherwise, let's keep it as is!